(2.22.0) Adds: missing query parameters

Ant-hem opened the pull request:
Fix #599

| Q                 | A
| ----------------- | ----------
| Bug fix?          | yes
| New feature?      | no
| BC breaks?        | no     
| Related Issue     | Fix #599
| Need Doc update   | no

aseure commented on algoliasearch-common/src/main/java/com/algolia/search/objects/QueryBase.java:
```diff
@@ -298,6 +303,63 @@
+                                .collect(Collectors.joining(","))
+                            + "]")
+                .collect(Collectors.joining(","))
+            + "]";
```
I don't know if this style is enforced by `coveo` but it is particularly difficult to read. May we try to change that?

Ant-hem replied:
It's coveo styling convention and I couldn't find a way to change it.

aseure reviewed (commented):
Great changes, thank you Antoine!

Only comment but maybe too late: we could have split this in multiple commits (one of each type added/changed and one for the tests) and some for the tests. This way, once we generate the ChangeLog entry, we could directly let users see what has changed (especially as we are introducing breaking changes). WDYT?

Ant-hem committed:
Adds: optional Filters

Ant-hem committed:
Adds: sumOrFiltersScores

Ant-hem committed:
Adds: similarQuery

Ant-hem committed:
Fixed: Legacy type parameters

Breaking change:
- numericFilters is now List<List<String>> instead of List<String>
- tagFilters is now List<List<String>> instead of List<String>
- analyticsTags is now List<String> instead of String.

This is also adding support for nested list when serializing the query
as urlParameters.

Ant-hem committed:
Tests: Updated with new parameters
